### PR TITLE
PR: Add clear button to find comboboxes

### LINKED ISSUE #5373: Add Clear button to (find) entry boxes
## Description of your problem

Recently I searched for a lot of strings – one after the other. They all came from an electronic document, so I could use copy'n'paste. Since I work on Linux, I just selected the search strings by double clicking it and wanted to paste them with a middle click into the find box. But the box was still populated with the search string from the previous search. This string had to be deleted first. Selecting it and middle click obviously does not work, so had to click into the entry box, switch to the keyboard (CTRL-A, DEL) and paste again with the mouse.

It would have been so much quicker if there were a "Clear search box" button as you can now see nearly everywhere.

Could you please provide such a button in every search box (I can come up with four: The _Editor_ has Find, Find + Replace, the _Find in files_ has one, _Help_ another, but there might be more.) Maybe other entry boxes can also gain from a Clear button?

## Versions and main components

* Spyder Version: 3.2.3 
* Python Version: 3.5.2 64bits
* Qt Version: 5.5.1
* PyQt Version: 5.5.1
* Operating system: Linux Mint 18.1

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:
IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  None (NOK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
numpy >=1.7      :  1.12.0 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
psutil >=0.3     :  5.3.1 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.2 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.10.7 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
sympy >=0.7.3    :  1.0 (OK)

### PR DESCRIPTION
**Pull request by timhoffm**
Fixes #5373

### CODE REVIEW

File: spyder/widgets/comboboxes.py [hunk 1]
```diff
@@ -135,6 +135,8 @@ class PatternComboBox(BaseComboBox):
     def __init__(self, parent, items=None, tip=None,
                  adjust_to_minimum=True):
         BaseComboBox.__init__(self, parent)
+        if hasattr(self.lineEdit(), 'setClearButtonEnabled'):  # only Qt >= 5.2
```
Comment (ccordoba12) [inline spyder/widgets/comboboxes.py]: Given that this only works for `Qt > = 5.2`, you need to enclose the lines you added in

```python
if PYQT5:
    ...
```

where `PYQT5` can be imported like this


```python
from qtpy import PYQT5
```

Comment (timhoffm) [reply spyder/widgets/comboboxes.py]: The `if hasattr()` serves exactly that purpose: Only use this feature if available. This should work without explicity checking a Qt version.

Also, I would assume that a simple `if PYQT5` would be `True` for  5.0 <= Qt < 5.2, but they still don't have the clear button.

Comment (ccordoba12) [reply spyder/widgets/comboboxes.py]: Sorry, you're right, `hasattr` should be enough.

Besides, we only support `Qt 5.2+`

**Comment by ccordoba12** [comment]
@dalthviz, please review that this is working locally for you.

File: spyder/widgets/comboboxes.py (hunk 1 above)
Comment (timhoffm) [reply spyder/widgets/comboboxes.py]: > Besides, we only support Qt 5.2+

In that case, just `self.lineEdit().setClearButtonEnabled(True)` would be enough.

The `if` is unneccesary clutter then. Should I remove it for brevity and clarity of the code?

Comment (ccordoba12) [reply spyder/widgets/comboboxes.py]: Nop, because we also support PyQt4.

**Comment by dalthviz** [comment]
@ccordoba12  I tested locally and worked for me :+1: (tested in PyQt4 and PyQt5)

**Comment by ccordoba12** [comment]
Thanks @dalthviz!

**Pull request merged**